api/src/API/Controllers/RacesController.cs

```diff
@@ -40,17 +43,78 @@ public async Task<IActionResult> GetAll()
             return Ok(result);
         }
 
+        [AllowAnonymous]
+        [HttpGet("public")]
+        public async Task<IActionResult> GetAllPublic()
+        {
+            RaceContainerClient container = containerProvider.RaceContainer;
+            IEnumerable<Race> result = await container.GetAllAsync();
+            return Ok(result);
+        }
+
         [HttpPost]
         public async Task<IActionResult> Create(Race race)
         {
             RaceContainerClient container = containerProvider.RaceContainer;
-            if (race.EventId == Guid.Empty)
+            if (!(await InitAndVerifyRace(race, container)))
             {
-                race.EventId = Guid.NewGuid();
+                return BadRequest();
             }
 
             await container.AddOneAsync(race);
             return CreatedAtAction(nameof(Create), new { id = race.Id }, race);
         }
+
+        [AllowAnonymous]
+        [HttpPost("public")]
+        public async Task<IActionResult> CreatePublic(Race race)
+        {
+            RaceContainerClient container = containerProvider.RaceContainer;
+            race.IsPublic = false;
+            if (!(await InitAndVerifyRace(race, container)))
+            {
+                return BadRequest();
+            }
+
+            await container.AddOneAsync(race);
+            return CreatedAtAction(nameof(Create), new { id = race.Id }, race);
+        }
+
+        [HttpPut]
```

Comment:
I'd prefer this to be a patch. I recognize I didn't implement it correctly on my end, but it'd be nice to send partial updates

Reply:
Are you sending partial updates from the FE?